go/apps/ctrl/services/build/backend/depot/generate_upload_url.go

```diff
@@ -0,0 +1,42 @@
+package depot
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"connectrpc.com/connect"
+
+	ctrlv1 "github.com/unkeyed/unkey/go/gen/proto/ctrl/v1"
+)
+
+func (s *Depot) GenerateUploadURL(
+	ctx context.Context,
+	req *connect.Request[ctrlv1.GenerateUploadURLRequest],
+) (*connect.Response[ctrlv1.GenerateUploadURLResponse], error) {
+	if req.Msg.UnkeyProjectId == "" {
+		return nil, connect.NewError(connect.CodeInvalidArgument,
+			fmt.Errorf("unkeyProjectID is required"))
+	}
+
+	// Generate unique S3 key for this build context
+	contextKey := fmt.Sprintf("%s/%d.tar.gz",
+		req.Msg.UnkeyProjectId,
+		time.Now().UnixNano())
```

Comment:
The concern I have is that this should imo deduplicate the same images.

if I do no changes and run deploy 3 times we are uploading 3 different tars, and building 3 seperate images. 

Thats not optimal tbh, when I press re-deploy with no changes we shouldn't re-build images but just use existing ones